tkupchyn/homework_03/next_happy_year.py

```diff
@@ -45,3 +44,18 @@ def next_happy_year(year: int) -> int:
                 is_happy_year = True
 
     return year
+
+
+def next_happy_year_alternative(year: int) -> int:
+    if year not in range(1000, 9001):
+        raise ValueError(f"Year {year} is out of range")
+
+    happy_year = len(set(str(year))) == 4
```

Comment:
you can use simple lambda not to repeat check logic in several places
```
is_happy_year = lambda year: len(set(str(year))) == 4

if is_happy_year(year):
    year += 1
```

Reply:
year = year + 1 if is_happy_year else year

Reply:
for increment you can use count

```
for i in count(year):
    if i is happy year:
         return
```

Reply:
Good point. Thanks